src/test/groovy/com/netflix/nebula/lint/rule/dependency/MultiProjectCircularDependencyRuleSpec.groovy

```diff
@@ -0,0 +1,264 @@
+                 compile 'joda-time:joda-time:latest.release\'
+                 compile 'org.ajoberstar:gradle-git:1.4.+\'        
+                 compile project(':bar')
+                 compile group: 'com.google.guava', name: 'guava', version: '23.5-jre'
+            }
+            '''.stripIndent())
+        addSubproject('bar', '''\
+            apply plugin: 'java'
+            dependencies {
+                 compile 'com.github.zafarkhaja:java-semver:latest.release\'
+                 compile 'com.google.guava:guava:20.0\'
+                 compile 'com.netflix.nebula:nebula-core:latest.release\'
+                 compile 'commons-lang:commons-lang:latest.release\'
+                 compile 'joda-time:joda-time:latest.release\'
+                 compile 'org.ajoberstar:gradle-git:1.4.+'   
+                 compile project(':foo')
+                 compile group: 'com.google.guava', name: 'guava', version: '23.5-jre' 
+            }
+            '''.stripIndent())
+
+
+        when:
+        def result = runTasksSuccessfully('fixGradleLint')
+
+        then:
+        result.standardOutput.contains("Multi-project circular dependencies are not allowed. Circular dependency found between projects 'foo' and 'bar'")
+    }
+
+    def 'detects circular dependencies in multi project - multi violations'() {
+        setup:
+        buildFile << """
+            plugins {
+                id 'java'
+            }
+            apply plugin: 'nebula.lint'
+
+            gradleLint.rules = ['multiproject-circular-dependency']    
+        """
+        addSubproject('foo', '''\
+            apply plugin: 'java'
+            dependencies {
+                 compile project(':bar')
+                 compile project(':baz')
+                 compile group: 'com.google.guava', name: 'guava', version: '23.5-jre'
+            }
+            '''.stripIndent())
+        addSubproject('bar', '''\
+            apply plugin: 'java'
+            dependencies {
+                compile project(':foo')
+                compile group: 'com.google.guava', name: 'guava', version: '23.5-jre' 
+            }
+            '''.stripIndent())
+        addSubproject('baz', '''\
+            apply plugin: 'java'
+            dependencies {
+                compile project(':foo')
+                compile group: 'com.google.guava', name: 'guava', version: '23.5-jre' 
+            }
+            '''.stripIndent())
+
+
+        when:
+        def result = runTasksSuccessfully('fixGradleLint')
+
+        then:
+        result.standardOutput.contains("Multi-project circular dependencies are not allowed. Circular dependency found between projects 'foo' and 'bar'")
+        result.standardOutput.contains("Multi-project circular dependencies are not allowed. Circular dependency found between projects 'foo' and 'baz'")
+    }
+
+    def 'detects circular dependencies in multi project - no parenthesis results in gradle failure but not lint'() {
```

Comment:
Is this something we can catch, as well?

Reply:
I guess we could, let me add a that to the rule -> invalid project declaration